subprojects/core-api/src/main/java/org/gradle/api/artifacts/ExternalModuleDependency.java

```diff
@@ -20,18 +20,19 @@
  */
 public interface ExternalModuleDependency extends ExternalDependency {
     /**
-     * Returns whether or not Gradle should always check for a change in the remote repository.
+     * Returns whether Gradle may check for a change in the remote repository
      *
      * @see #setChanging(boolean)
      */
     boolean isChanging();
 
     /**
-     * Sets whether or not Gradle should always check for a change in the remote repository. If set to true, Gradle will
-     * check the remote repository even if a dependency with the same version is already in the local cache. Defaults to
-     * false.
+     * Sets whether Gradle may check for a change in the remote repository. If set to true, Gradle will
+     * check the remote repository if the local entry has expired.
```

Comment:
isnt it that wrong?
if changing is true than the local entry will get an expiry date and it will be checked after that. if its not changing it wont expire and hence wont be checked until manually deleted or --update-dependencies. At least thats how i understood it

Reply:
Yes, changing dependencies get checked at timed interval which means it can be refreshed even if there is a local cached entry.